### PR TITLE
snapshot/restore documentation

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

Long overdue howto for snapshot/restore

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by aglitke — COMMENTED** [review]
Looks good to me.  Nice and simple.

File: docs/snapshot-restore.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+```bash
+kubectl patch -n kubevirt kubevirt kubevirt -p '{"spec": {"configuration": { "developerConfiguration": { "featureGates": [ "Snapshot" ] }}}}' -o json --type merge
+```
+
+## Snapshot a VirtualMachine
+
+To snapshot a `VirtualMachine` named `larry`, apply the following yaml.
+
+\* It is currently a requirement that `VirtualMachines` are halted before taking a snapshot.
```
Comment (arnongilboa) [inline docs/snapshot-restore.md:33]: halt means pause or stop?

File: docs/snapshot-restore.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+Make sure that the `provisioner` property of your `StorageClass` matches the `driver` property of the `VolumeSnapshotClass`
+
+Even if you have no `VlumeSnapshotClasses` in your cluster, `VirtualMachineSnapshots` are not totally useless.  They will still backup your `VirtualMachine` configuration.
+
+### Snapshot Feature Gate
+
+Snapshot/Restore are currently considered alpha features and are disabled by default.
+
```
Comment (arnongilboa) [inline docs/snapshot-restore.md:24]: To enable it:

File: docs/snapshot-restore.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+To list `VolumeSnapshotClasses`:
+
+```bash
+kubectl get volumesnapshotclass
+```
+
+Make sure that the `provisioner` property of your `StorageClass` matches the `driver` property of the `VolumeSnapshotClass`
+
+Even if you have no `VlumeSnapshotClasses` in your cluster, `VirtualMachineSnapshots` are not totally useless.  They will still backup your `VirtualMachine` configuration.
```
Comment (ngavrilo) [inline docs/snapshot-restore.md:19]: VlumeSnapshotClasses missing an 'o'

File: docs/snapshot-restore.md (hunk 1 above)
Comment (ngavrilo) [inline docs/snapshot-restore.md:33]: I'm also not sure if a user would understand that we mean to stop and not just to pause

**Pull request merged**